x-pack/test/functional/services/ml/settings_calendar.ts

```diff
@@ -9,71 +9,26 @@ import expect from '@kbn/expect';
   { getService }: FtrProviderContext,
-  mlCommonUI: MlCommonUI,
-  navigation: Navigation,
-  jobTable: MlADJobTable
+  mlCommonUI: MlCommonUI
 ) {
   const testSubjects = getService('testSubjects');
   const retry = getService('retry');
   const comboBox = getService('comboBox');
 
   return {
-    async checkForCalendar(jobNumbers: number[], checkForExists: boolean = true) {
-      for await (const jobNumber of jobNumbers)
-        await jobTable.withDetailsOpen(`test_calendar_ad_${jobNumber}`, async () => {
-          await testSubjects[checkForExists ? 'existOrFail' : 'missingOrFail'](
-            jobTable.detailsSelector(
-              `test_calendar_ad_${jobNumber}`,
-              'mlJobRowDetailsSection-calendars'
-            )
-          );
-        });
-    },
-
-    async assertConnectedJobs(calendarId: string, regexp: RegExp) {
-      // Assert that the Calendar Management view shows that the calendar is only connected to the jobs
-      // applied at creation
-      await this.assertCalendarRowJobs(calendarId, regexp);
-
-      await navigation.navigateToAnomalyDetection();
-
-      // withDetailsOpen will open up the details of the specific job, and then collapse the details section afterwards
-      await this.checkForCalendar([1, 2]);
-
-      await this.checkForCalendar([3, 4], false);
-
-      await testSubjects.click('mlMainTab settings');
-      await testSubjects.click('mlCalendarsMngButton');
-      await testSubjects.click('mlEditCalendarLink');
+    async assertCalendarRowJobs(calendarId: string, expectedConnectedJobs: string[]) {
+      await this.filterWithSearchString(calendarId, 1); // makes sure the table only has the one row we want to check
+      const calendarRow = (await this.parseCalendarTable())[0];
+      const actualJobs: string[] = calendarRow.jobs.replaceAll(',', '').split(' ');
 
-      await this.selectJobGroup('multi-metric');
-      await this.saveCalendar();
-
-      const groupsAfterAddingMultiMetricsVisibleText = await testSubjects.getVisibleText(
-        'mlCalendarListColumnJobs'
+      const expectedWithinActual = expectedConnectedJobs.every((expected) =>
+        actualJobs.some((actual) => actual === expected)
       );
-      expect(groupsAfterAddingMultiMetricsVisibleText).to.match(/multi-metric/);
-
-      // Go back to the Anomaly Detection Jobs view
-      await navigation.navigateToAnomalyDetection();
-
-      // Assert that the calendar is now connected to the multimetric job group
-      await this.checkForCalendar([4]);
 
-      // Assert that the calendar is still not connected to the automated job group
-      await this.checkForCalendar([3], false);
-    },
-
-    async assertCalendarRowJobs(calendarId: string, regexp: RegExp) {
-      const rowJobsListVisibleText = await testSubjects.getVisibleText(
-        `mlCalendarListRow row-${calendarId}`
-      );
-      expect(rowJobsListVisibleText).to.match(regexp);
+      expect(expectedWithinActual).to.be(true);
```

Comment:
This sort of assertion is not ideal as it gives basically no hint about what was going wrong in case of a failure (in this case the failure message would be something like `expected false to equal true`.

Sometimes, the expect library doesn't have the proper assertion that we want to do and we have to fallback to some pre-expect evaluation (like you did here). For these cases, we could use an extra validation message that's supported by most expect methods (but `be` is not in that list), e.g. like this
```
expect(expectedWithinActual).to.eql(true, `Calendar ${calendarId} should be assigned to jobs ${expectedConnectedJobs}, got ${actualJobs}`)
```

Reply:
[Here](https://github.com/elastic/kibana/pull/178553/files#diff-1ea44821f5dd58001f0f78f15f6d6bfd5c9ce22ca83f9b4fb7b175a08d1aea92R39) we go boss